unix_integration/src/unix_proto.rs

```diff
@@ -30,28 +30,34 @@ pub struct DeviceAuthorizationResponse {
     pub message: Option<String>,
 }
 
+#[derive(Serialize, Deserialize, Debug)]
+pub enum MFAType {
+    Code, // PAM must prompt for an authentication code
+    Poll, // PAM can immediately return. The provider will poll for an external response
```

Comment:
Should poll be part of the clients request? Because else how does the pam module go "idle" waiting for the resolver to respond?

Reply:
The poll is happening in the id provider. I only named it 'Poll' meaning what the id provider is doing. I guess that's confusing. The Poll type is operating the same as the DeviceAuthorizationGrant, which is, print a prompt, then immediately call the daemon and wait (the reason for the long timeout) for it to acknowledge the auth completed.

Reply:
I think I've resolved this one by modeling the diagram you provided.